metrics/cmd/heapster-apiserver/app/server.go

```diff
@@ -60,12 +60,11 @@ func NewHeapsterApiServer(s *options.HeapsterRunOptions, metricSink *metricsink.
 }
 
 func newAPIServer(s *options.HeapsterRunOptions) (*genericapiserver.GenericAPIServer, error) {
-	if err := s.SecureServing.MaybeDefaultWithSelfSignedCerts("heapster.kube-system"); err != nil {
+	if err := s.SecureServing.MaybeDefaultWithSelfSignedCerts("heapster.kube-system", nil, nil); err != nil {
```

Comment:
Consider removing support for aggregated apiserver as this is currently implemented by metrics-server.

@DirectXMan12 WDYT?

Reply:
I think it's OK to keep it in this PR. Let's not try to overcomplicate it.